actors.md

```diff
@@ -1144,14 +1215,14 @@ TODO: Define me
 type UpdateChannelState struct {
   sv SignedVoucher
   secret Bytes
-  pip PieceInclusionProof
+  proof Bytes
 } representation tuple
 ```
 
 **Algorithm**
 
 ```go
-func UpdateChannelState(sv SpendVoucher, secret []byte, pip *PieceInclusionProof) {
+func UpdateChannelState(sv SignedVoucher, secret []byte, proof []byte) {
```

Comment:
See comment above about who provides the sector ID.

I don't think a single proof supplied by the channel target is sufficient in this case, and I'm much less certain that it will suffice for all future conditional payments.

* One alternative would be to make the third channel an `interface{}` style parameter that would accept a proof struct. This will provide the most (runtime) type safety, but means a larger ABI interface.
* Another would be to accept a `[]interface{}` that would contain a sector id and proof in this case.
* A third would be to require a `proof` to be sufficient, but that would require some messy encoding in the voucher creation and decoding in the verification actor.

Reply:
`proof` here is anything, it can be a serialized struct, it can actually be bytes, its just something that the miner creates and uses to prove that they can cash in the payment channel.

So effectively, its an `interface{}`, but since an empty interface is a go thing, a serialized byte array is the next best thing.